### PR TITLE
STY: Address or suppress most pylint complaints

### PR DESCRIPTION
**Pull request by tanvimoharir**
Fixes for the following which come up in _version.py:
1.unused-variable i (src.from_parentdir,src.long_get_versions)
2.no-else-return(src.from_parentdir)
3.unused-variable out (src.from_vcs)
4.redefined-outer-name run_command (src.from_vcs,test.git.test_git)

### CODE REVIEW

File: src/git/from_vcs.py
Comment (effigies) [inline src/git/from_vcs.py:17]: FWIW I don't think `r_c` is clearer than `run_command`. What issue is being resolved by this one?

Comment (tanvimoharir) [reply src/git/from_vcs.py:17]: /_version.py:219:51: W0621: Redefining name 'run_command' from outer scope (line 70) (redefined-outer-name)
Yeah I agree r_c is not clear, I'm still thinking about this.

Comment (bryevdv) [reply src/git/from_vcs.py:17]: Offhand suggestion (I am not well-versed in the details): rename the outer function to `default_run_command`

Comment (tanvimoharir) [reply src/git/from_vcs.py:17]: What precisely do you mean by 'suppressed in a single place'?

Comment (effigies) [reply src/git/from_vcs.py:17]: I think @bryevdv meant adding something like `#noqa` to this line, or some other marker that would hopefully avoid the problem.

What about:

```suggestion
def git_pieces_from_vcs(tag_prefix, root, verbose, runner=run_command):
```

**Comment by tanvimoharir** [comment]
Hi,
I have fixed whatever changes were possible.
Most of the changes were 'invalid-name' and I modified the name wherever I could and suppressed them for few( did not rename the variable 'VCS' since the impact seemed more and for other variables I could not derive context so I did not rename them)

**Comment by effigies** [comment]
@tanvimoharir Sorry, I think I looked at this, saw the tests were failing, but didn't look into it. It seems that there are flake8 issues with the source. Would you mind cleaning those up? I'll have a final review once that's done.

**Comment by tanvimoharir** [comment]
> @tanvimoharir Sorry, I think I looked at this, saw the tests were failing, but didn't look into it. It seems that there are flake8 issues with the source. Would you mind cleaning those up? I'll have a final review once that's done.

Sure , I will take a look at it.

**Comment by effigies** [comment]
Had a look myself. Applied the suggestion from https://github.com/python-versioneer/python-versioneer/pull/250#discussion_r604870865 and addressed the style complaints. We'll see if tests are happy...

**Comment by tmoharir** [comment]
> Had a look myself. Applied the suggestion from [#250 (comment)](https://github.com/python-versioneer/python-versioneer/pull/250#discussion_r604870865) and addressed the style complaints. We'll see if tests are happy...

Oh okay,I had added # noqa: E127,E128,W291 for the flake issues and pylint suppressions for each of them.
Was going to merge with upstream branch as well(since tests were failing)

**Comment by tanvimoharir** [comment]
> Had a look myself. Applied the suggestion from [#250 (comment)](https://github.com/python-versioneer/python-versioneer/pull/250#discussion_r604870865) and addressed the style complaints. We'll see if tests are happy...

Oh okay i had added # noqa: E127,E128,W291 for the flake issues and pylint suppressions for each of these( and few more)

**Review by effigies — APPROVED** [review]
LGTM, assuming tests pass. Did you have remaining tasks you wanted to do here?

**Comment by tanvimoharir** [comment]
> LGTM, assuming tests pass. Did you have remaining tasks you wanted to do here?

No I don't have any remaining tasks.

**Pull request merged**
